ADLStream/data/preprocessing/standardization_scaler.py

```diff
@@ -0,0 +1,138 @@
+"""Implements a mean normalization scaler"""
+from ADLStream.data.preprocessing import BasePreprocessor
+import math
+
+
+class StandardizationScaler(BasePreprocessor):
+    """The transformation is given by
+        x_scaled = (x - avg_x) / x_stdev
+
+    where avg_x is the mean seen until now for the feature x and x_stdev
+    is the standard deviation seen until now for teh feature x.
+
+    Arguments:
+        share_params (bool): Whether to share scaler parameters among columns.
+            Defaults to False.
+
+    """
+
+    def __init__(self, share_params=False):
+        self.share_params = share_params
+        self.data = None
+        self.data_sum = None
+        self.data_stdev_sum = None
+        self.data_avg = None
+        self.data_stdev = None
+        self.data_count = 1
+
+    def _mean(self, a):
+
+        if self.share_params == False and self.data_count != 1:
+            assert len(a) == len(self.data_sum)
+            self.data_sum = [self.data_sum[i] + a[i] for i in range(len(a))]
+            mean = [(self.data_sum[i]) / self.data_count for i in range(len(a))]
+
+        elif self.share_params == False and self.data_count == 1:
+            assert len(a) == len(self.data_sum)
+            mean = [(self.data_sum[i]) / self.data_count for i in range(len(a))]
+
+        elif self.share_params == True and self.data_count != 1:
+            self.data_sum += sum(a)
+            mean = [self.data_sum / (self.data_count * len(a))] * len(a)
+
+        elif self.share_params == True and self.data_count == 1:
+            mean = [self.data_sum / (self.data_count * len(a))] * len(a)
+        return mean
+
+    def _standard_deviation(self, a):
+        if self.share_params == False:
+            assert len(a) == len(self.data_sum)
+        mean = self.data_avg
+        if self.share_params == False and self.data_count != 1:
+            self.data.append(a)
```

Comment:
The `self.data` will contain the complete history of the stream. Note that this is not allowed in a streaming scenario. 
We need to rethink how to compute the standard deviation in an online manner.

As a first approach I'll suggest something like this:

> Given that the standard deviation can be defined as
> ![formula](https://render.githubusercontent.com/render/math?math={\large\color{grey}\sigma%20=%20\sqrt{\bar{x^2}%20-%20(\bar{x})^2}%20=%20\sqrt{\frac{\sum{x^2}}{N}%20-%20\left(\frac{\sum{x}}{N}\right)^2}})
> We will just need to keep three parameters:
> 1. `sum1 = ` The sum of the values: ![formula](https://render.githubusercontent.com/render/math?math={\color{grey}\sum{x}}&mode=inline)
> 2. `sum2 = ` The sum of the squared values: ![formula](https://render.githubusercontent.com/render/math?math={\color{grey}\sum{x^2}}&mode=inline)
>3. `count = ` The number of instances ![formula](https://render.githubusercontent.com/render/math?math={\color{grey}N}&mode=inline)
> 
> Then std is `std = sqrt( (sum2/count) - ((sum1/count))**2)`

This is just one way, butthere exist several alternatives out there. 
Please, have a look at these algorithms https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
Should we implement **Welford's online algorithm** here?

Reply:
In addition, we probably do not need the `self.data_count == 1` special case. (see `_mean` comment)